extension/export_Tensor.cc

```diff
@@ -122,7 +122,7 @@ static py::array Tensor_dot_list(const Tensor& self, py::list tensors) {
   py::array_t<scalar_type> ret(dots.size());
   std::copy(dots.begin(), dots.end(), ret.mutable_data());
 
-  return ret;
+  return std::move(ret);
```

Comment:
Are you sure this is correct? That looks very weird to me and breaks a couple of patterns I am aware of.

Reply:
Apple Clang tells me the following:
```
extension/export_Tensor.cc:125:10: error: local variable 'ret' will be copied despite being returned by name [-Werror,-Wreturn-std-move]
  return ret;
         ^~~
extension/export_Tensor.cc:125:10: note: call 'std::move' explicitly to avoid copying
  return ret;
         ^~~
         std::move(ret)
```

We can stick to the previous version (without `std::move`) and disable the warning.

Reply:
Hmmm ... I think this might be a complicated matter and ignoring it is not the best solution. Understanding is better. 

See for example this [stackoverflow](https://stackoverflow.com/questions/4986673/c11-rvalues-and-move-semantics-confusion-return-statement) thread. If I understand correctly, the move-constructor of `array_t` should have been automatically called, so question is why not?

Reply:
Oh boy... that sounds complicated.
Do we need to check whether `array_t` is move-constructible? Maybe you can have a [look](https://github.com/pybind/pybind11/blob/master/include/pybind11/numpy.h)... I have basically no experience with such things...

Reply:
I found the issue, see https://reviews.llvm.org/D43322 ... the problem is the return type is different! Just replace `std::array` by `std::array<scalar_type>` and that's it.

Reply:
Done.